src/main/java/io/jenkins/plugins/propelo/job_reporter/extensions/LevelOpsRunListener.java

```diff
@@ -187,15 +188,32 @@ public void onFinalized(Run run) {
         }
     }
 
+    /**
+     * Deletes data transmitted to propelo
+     */
     private void deleteJobRunDataCompleteDirectoryContents() {
+        LOGGER.log(Level.FINEST, "Delete Job Run Complete Data Directory (date-$date) starting");
         File dataDirectoryWithVersion = plugin.getDataDirectoryWithVersion();
-        if (dataDirectoryWithVersion != null && dataDirectoryWithVersion.exists()) {
-            LOGGER.log(Level.FINEST, "Delete Job Run Complete Data Directory (date-$date) starting");
-            for (File file : Objects.requireNonNull(dataDirectoryWithVersion.listFiles())) {
-                if (!file.getName().equalsIgnoreCase(DateUtils.getDateFormattedDirName())) {
-                    FileUtils.deleteQuietly(file);
-                }
-            }
+        if (dataDirectoryWithVersion == null || !dataDirectoryWithVersion.exists()) {
+            LOGGER.log(Level.FINE, "Skipping job run complete data directory delete... data directory = {0}", dataDirectoryWithVersion);
+            return;
+        }
+        String todayDirName = DateUtils.getDateFormattedDirName();
+        try {
```

Comment:
Yes, that's the goal but, this is impacting users and the change has been run through integration testing.
Unit testing it is not straight forward since the original code base doesn't have a test for the original use and since this is a private method, setting up testing will take longer.
So, we will add the missing unit tests for the module to prevent regressions while we release the fix and solve the user's issues.